### PR TITLE
ADEN-2418 Don't accept monocolor ad

### PR DESCRIPTION
**Pull request by drets**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 4 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Remove this empty statement.](https://github.com/Wikia/selenium-tests/blob/c875b84765af9e16d659ad0111c977cab910b0b8/src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageComparison.java#L62) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/selenium-tests/blob/c875b84765af9e16d659ad0111c977cab910b0b8/src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageComparison.java#L95) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/selenium-tests/blob/c875b84765af9e16d659ad0111c977cab910b0b8/src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageComparison.java#L33) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Define a constant instead of duplicating this literal "ScreenshotsComparison" 3 times.](https://github.com/Wikia/selenium-tests/blob/c875b84765af9e16d659ad0111c977cab910b0b8/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L77) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/c875b84765af9e16d659ad0111c977cab910b0b8/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L67) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MINOR [Define a constant instead of duplicating this literal "verifyColorAd" 3 times.](https://github.com/Wikia/selenium-tests/blob/c875b84765af9e16d659ad0111c977cab910b0b8/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L124) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)

File: src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageComparison.java [hunk 1]
```diff
@@ -74,6 +73,17 @@ public boolean isColorImage(BufferedImage image, Color color, int accuracy) {
     return true;
   }
 
+  public boolean isMonocolorImage(BufferedImage image) {
```
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageComparison.java]: Does this method return true, if given image has only one colour in its entire area ?
Please add a javadoc that clarifies that

Comment (drets) [inline src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageComparison.java]: :+1:

**Comment by justnpT** [comment]
lgtm

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 3 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/selenium-tests/blob/24960089c0480a377fd37693686831e4b23797cc/src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageComparison.java#L33) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/selenium-tests/blob/24960089c0480a377fd37693686831e4b23797cc/src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageComparison.java#L97) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Define a constant instead of duplicating this literal "verifyColorAd" 3 times.](https://github.com/Wikia/selenium-tests/blob/24960089c0480a377fd37693686831e4b23797cc/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L124) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/24960089c0480a377fd37693686831e4b23797cc/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L67) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MINOR [Define a constant instead of duplicating this literal "ScreenshotsComparison" 3 times.](https://github.com/Wikia/selenium-tests/blob/24960089c0480a377fd37693686831e4b23797cc/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java#L77) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)

**Pull request merged**